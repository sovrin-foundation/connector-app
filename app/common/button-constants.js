// @flow
export const DENY = 'Deny'
export const CONNECT = 'Connect'
