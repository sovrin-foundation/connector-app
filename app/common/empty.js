// @flow
// sometimes we need an empty object to use as a placeholder in destructuring
export default {}
