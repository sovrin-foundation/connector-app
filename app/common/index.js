// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
