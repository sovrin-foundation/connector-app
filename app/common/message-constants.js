// @flow
// for overall app message string, it can be error, success, warning etc.

export const TOUCH_ID_MESSAGE = 'Please confirm with Biometrics'
export const TOUCH_ID_NOT_AVAILABLE =
  'Biometrics is not supported on this device'

export const DEVICE_ENROLLMENT_ERROR = [
  'Enrollment Error',
  'Identifier or Seed or Push-notification token not present!',
]

export const PUSH_NOTIFICATION_PERMISSION_ERROR = [
  'Permission Error',
  'Push Notification permission needed! Please enable it from Settings.',
]

export const DUPLICATE_CONNECTION_ERROR = [
  'Attention!',
  'Duplicate Connection invitation!',
]

export const UNLOCKING_APP_WAIT_MESSAGE = 'Loading please wait...'

export const ENTER_PASS_CODE_MESSAGE = 'Enter passcode'

export const ENTER_YOUR_PASS_CODE_MESSAGE = 'Enter your passcode'

export const FIRST_BACKUP_TITLE = 'Set up data recovery now!'

export const FIRST_BACKUP_SUBTEXT = '2 easy steps, takes 3 minutes.'

export const SUBSEQUENT_BACKUP_TITLE =
  'You have new data that is not backed up.'

export const SUBSEQUENT_BACKUP_SUBTEXT =
  "Create another backup so you don't lose anything."

export const PASSPHRASE_GENERATION_ERROR =
  'Uh oh... Looks like something went wrong.'
