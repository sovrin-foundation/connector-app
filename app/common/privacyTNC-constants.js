// @flow

import { Platform } from 'react-native'

const isAndroid = Platform.OS === 'android'
export { EULA_URL as TermsAndConditionUrl } from '../eula/type-eula'
export const TermsAndConditionsTitle = 'Terms and Conditions'

export const PrivacyPolicyUrl = 'https://www.connect.me/privacy.html'
export const localPrivacyPolicySource = isAndroid
  ? 'file:///android_asset/external/connectme/privacy.html'
  : './privacy.html'
export const PrivacyPolicyTitle = 'Privacy Policy'
