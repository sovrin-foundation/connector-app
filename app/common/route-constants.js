// @flow
export const splashScreenRoute = 'SplashScreenHide'
export const homeRoute = 'Home'
export const authenticationRoute = 'Authentication'
export const connectionRoute = 'Connection'
export const settingsRoute = 'Settings'
export const settingsScreen = 'SettingsStack'
export const expiredTokenRoute = 'ExpiredToken'
export const qrCodeScannerRoute = 'QRCodeScanner'
export const lockPinSetupRoute = 'PinSetup'
export const lockTouchIdSetupRoute = 'TouchIdSetup'
export const lockPinSetupHomeRoute = 'PinSetupHome'
export const lockSelectionRoute = 'LockSelection'
export const lockEnterPinRoute = 'LockEnterPin'
export const lockSetupSuccessRoute = 'LockSetupSuccess'
export const lockEnterFingerprintRoute = 'LockEnterFingerprint'
export const claimOfferRoute = 'ClaimOffer'
export const homeTabRoute = 'TabHome'
export const connectionsTabRoute = 'TabHome'
export const credentialsTabRoute = 'TabCredentials'
export const discoverTabRoute = 'TabDiscover'
export const menuTabRoute = 'TabMenu'
export const settingsTabRoute = 'TabSettings'
export const qrCodeScannerTabRoute = 'TabQRCodeScanner'
export const proofRequestRoute = 'ProofRequest'
export const invitationRoute = 'InvitationRoute'
export const connectionHistoryRoute = 'ConnectionHistoryRoute'
export const connectionHistoryDetailsRoute = 'ConnectionHistoryDetailsRoute'
export const switchEnvironmentRoute = 'SwitchEnvironment'
export const lockAuthorizationRoute = 'LockAuthorizationRoute'
export const lockAuthorizationHomeRoute = 'LockAuthorizationHomeRoute'
export const waitForInvitationRoute = 'WaitForInvitationRoute'
export const connectionHistoryPendingRoute = 'ConnectionHistoryPendingRoute'
export const walletRoute = 'TokenWallet'
export const walletTabSendDetailsRoute = 'WalletTabSendDetails'
export const receiveTabRoute = 'WalletReceiveTab'
export const sendTabRoute = 'WalletSendTab'
export const historyTabRoute = 'WalletHistoryTab'
export const aboutAppRoute = 'AboutApp'
export const onfidoRoute = 'Onfido'
export const privacyTNCRoute = 'PrivacyTNC'
export const eulaRoute = 'EulaScreen'
export const genRecoveryPhraseRoute = 'GenerateRecoveryPhrase'
export const verifyRecoveryPhraseRoute = 'VerifyRecoveryPhrase'
export const exportBackupFileRoute = 'ExportBackupFile'
export const backupCompleteRoute = 'BackupComplete'
export const restoreRoute = 'RestoreStartScreen'
export const restoreWaitRoute = 'RestoreWaitScreen'
export const restorePassphraseRoute = 'RestorePassphrase'
export const backupErrorRoute = 'BackupErrorScreen'
