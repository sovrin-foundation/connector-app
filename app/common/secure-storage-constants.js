// @flow

export const PUSH_COM_METHOD = 'pushComMethod'
export const CONNECT_ME_STORAGE_KEY = '@connectmeDefaults'
export const IS_ALREADY_INSTALLED = `${CONNECT_ME_STORAGE_KEY}:IS_ALREADY_INSTALLED`
export const CONNECTIONS = 'CONNECTIONS'
export const CLAIM_MAP = `CLAIM_MAP`
export const WALLET_ENCRYPTION_KEY = 'walletEncryptionKey'
export const WALLET_BALANCE = 'walletBalance'
export const WALLET_ADDRESSES = 'walletAddresses'
export const WALLET_HISTORY = 'walletHistory'
export const LAST_SUCCESSFUL_BACKUP = 'lastSuccessfulBackup'
export const PASSPHRASE_SALT_STORAGE_KEY = 'passphraseSalt'
export const PASSPHRASE_STORAGE_KEY = 'passphrase'
