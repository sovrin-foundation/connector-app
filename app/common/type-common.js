// @flow

import type { Element } from 'react'
import type { NavigationScreenProp } from 'react-navigation'

export type CustomError = {
  code: string,
  message: string,
}

export const INITIAL_TEST_ACTION: 'INITIAL_TEST_ACTION' = 'INITIAL_TEST_ACTION'

export type InitialTestAction = {
  type: typeof INITIAL_TEST_ACTION,
}

export const initialTestAction = () => ({
  type: INITIAL_TEST_ACTION,
})

export type NavigationParams = {
  [string]: any,
}
export type NavigationRoute = NavigationLeafRoute
export type NavigationState = {
  index: number,
  routes: Array<NavigationRoute>,
}

export type NavigationLeafRoute = {|
  key: string,
  routeName: string,
  path?: string,
  params?: NavigationParams,
|}

export type ReactNavigation = NavigationScreenProp

export type GenericObject = {
  [string]: any,
}

export type GenericStringObject = {
  [string]: string,
}

export type MessageAnnotation = {
  name: string,
  version: string,
}

export type TopicAnnotation = {
  tid: number,
  mid: number,
}

export type ReactChildren = Element<*>

export const RESET: 'RESET' = 'RESET'
export const REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS: 'REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS' =
  'REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS'

export type ResetAction = {
  type: typeof RESET,
}

export type ImageSource = {
  uri: string,
}

export type NotificationPayload = {
  forDID: string,
  uid: string,
  type: string,
  remotePairwiseDID: string,
  senderLogoUrl?: ?string,
}

export type ImmutableGenericStringObject = {
  +[string]: string,
}

export const STORE_STATUS = {
  IDLE: 'IDLE',
  IN_PROGRESS: 'IN_PROGRESS',
  ERROR: 'ERROR',
  SUCCESS: 'SUCCESS',
}

export type StoreError = { +error: ?CustomError }
export type StoreStatus = { +status: $Keys<typeof STORE_STATUS> }

export type StatusBarStyle = 'default' | 'light-content' | 'dark-content'
