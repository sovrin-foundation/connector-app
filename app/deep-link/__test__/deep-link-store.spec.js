// @flow
import deepLinkReducer, {
  deepLinkData,
  deepLinkEmpty,
  deepLinkError,
} from '../deep-link-store'
import { RESET, initialTestAction } from '../../common/type-common'

describe('Deep link store', () => {
  let initialState
  beforeEach(() => {
    initialState = deepLinkReducer(undefined, initialTestAction())
  })

  it('should correctly update store when we get deep link data', () => {
    expect(
      deepLinkReducer(initialState, deepLinkData('erg76sd'))
    ).toMatchSnapshot()
  })

  it('should update store for empty link', () => {
    expect(deepLinkReducer(initialState, deepLinkEmpty())).toMatchSnapshot()
  })

  it('should update store for error', () => {
    expect(
      deepLinkReducer(initialState, deepLinkError('error'))
    ).toMatchSnapshot()
  })

  it('should reset, if RESET action is raised', () => {
    const afterDeepLinkDataState = deepLinkReducer(
      initialState,
      deepLinkData('erg76esd')
    )
    expect(
      deepLinkReducer(afterDeepLinkDataState, { type: 'RESET' })
    ).toMatchSnapshot()
  })
})
