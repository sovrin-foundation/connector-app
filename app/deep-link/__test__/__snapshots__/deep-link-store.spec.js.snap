// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deep link store should correctly update store when we get deep link data 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "tokens": Object {
    "erg76sd": Object {
      "error": null,
      "status": "NONE",
      "token": "erg76sd",
    },
  },
}
`;

exports[`Deep link store should reset, if RESET action is raised 1`] = `
Object {
  "error": null,
  "isLoading": true,
  "tokens": Object {},
}
`;

exports[`Deep link store should update store for empty link 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "tokens": Object {},
}
`;

exports[`Deep link store should update store for error 1`] = `
Object {
  "error": "error",
  "isLoading": false,
  "tokens": Object {},
}
`;
